backend/src/modules/student/student_entity.py

```diff
@@ -54,10 +48,10 @@ def to_dto(self) -> "Student":
         """Convert entity to DTO using the account relationship."""
         return Student(
             id=self.account_id,
-            pid=str(self.account_id),
+            pid=self.account.pid or str(self.account_id),
```

Comment:
Cascading change from pid not being nullable as I mentioend above: we can remove the `or` condition here
```suggestion
            pid=self.account.pid,
```